rust/ip-packet/src/lib.rs

```diff
@@ -1409,9 +1410,9 @@ impl<'a> IpPacket<'a> {
     }
 
     /// Unwrap this [`IpPacket`] as a DNS message, panicking in case it is not.
-    pub fn unwrap_as_dns(&self) -> hickory_proto::op::Message {
+    pub fn unwrap_as_dns(&self) -> Message<Vec<u8>> {
```

Comment:
Unfortunately, that doesn't work because `payload` doesn't track the lifetime of the underlying buffer. See https://github.com/libpnet/libpnet/issues/319.